docs/src/main/asciidoc/ghpages.sh

```diff
@@ -12,6 +12,40 @@ if ! [ -d docs/target/generated-docs ]; then
     exit 0
 fi
 
+# The script should be executed from the root folder
+
+ROOT_FOLDER=`pwd`
+echo "Current folder is ${ROOT_FOLDER}"
+
+if [[ ! -e "${ROOT_FOLDER}/.git" ]]; then
+    echo "You're not in the root folder of the project!"
+    exit 1
+fi
+
+# Retrieve version number, name of the main adoc and name of the current branch
+###################################################################
+
+# Code grepping for the 1st presence of "version>" in pom.xml.
+# First one is project version, second parent version.
+VERSION_NODE=`awk '/version>/{i++}i==1{print; exit}' $ROOT_FOLDER/pom.xml`
+# Extract the contents of the version node
+VERSION_VALUE=$(sed -ne '/version/{s/.*<version>\(.*\)<\/version>.*/\1/p;q;}' <<< "$VERSION_NODE")
+echo "Extracted version from root pom.xml is [${VERSION_VALUE}]"
+
+# Code grepping for the 2nd presence of "version>" in pom.xml.
+# First one is parent, second project version.
```

Comment:
Ok so we don't want to version the lib in the way Spring is versioned for instance? Don't we want to be consistent?(e.g. http://docs.spring.io/spring-boot/docs/ ) . They have a folder for each release / snapshot version and that's pretty nice for someone who is using a particular version of the docs, isn't it?

If not then achieving what you say will be quite trivial. We'd have to change the code for master so that we don't publish docs to an additional folder. For the rest we don't use the version at all but choose the branch name as the name of the folder. Our Jekins would call the script for 1.0.x and master.